File: selfstudy/js/students_reverse.js
/*
task 18:
Students - Reverse
You need to reverse the order of student test scores
*/

function reverse_order_students(scores) {
  return [...scores].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_students([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_students([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_students([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_students([100, 200]) },
  { input: [], expected: reverse_order_students([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_students', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_students(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_students(t.input) === t.expected || JSON.stringify(reverse_order_students(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think

File: selfstudy/golang/students_reverse.go
/*
task 218:
Students - Reverse
You need to reverse the order of student test scores
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderStudents(scores []int) []int {
	res := make([]int, len(scores))
	for i, v := range scores {
		res[len(scores)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderStudents([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderStudents([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderStudents([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderStudents([]int{100, 200})},
		{[]int{}, Reverse_orderStudents([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderStudents(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
